File: Problem16/PowerDigitSum.rb
total = 1
for n in 0..999 do
    total *= 2    
end

total_array = total.to_s.split('').map(&:to_i)
# puts total_array 
sum = 0
# puts total_array.length()
for i in 0..total_array.length()-1 do
   sum += total_array[i]
end

puts sum

File: Problem16/PowerDigitSum.js
// // 2^15 = 32768 and the sum of its digits is 3 + 2 + 7+ 6 + 8 = 26
// What is the sum of the digits of the number 2^1000?
const PowerDigitSum = ()=>{

    let total = 1n
    
    for(let i = 0; i < 1000; i++){
        total *= 2n
    }
    let totalArray = total.toString()
    let sum = 0n
    for(let i = 0; i < totalArray.length; i++){
        // console.log(totalArray[i])
        sum+=BigInt(parseInt(totalArray[i])) 
    }
    // console.log(sum)
    // console.log(totalArray)
    return sum
}

module.exports = PowerDigitSum